apps/snitch_api/lib/snitch_api_web/views/product_view.ex

```diff
@@ -0,0 +1,33 @@
+defmodule SnitchApiWeb.ProductView do
+  use SnitchApiWeb, :view
+  use JaSerializer.PhoenixView
+
+  location("/products/:slug")
+
+  attributes([
+    :name,
+    :description,
+    :available_on,
+    :deleted_at,
+    :discontinue_on,
+    :slug,
+    :meta_description,
+    :meta_keywords,
+    :meta_title,
+    :promotionable
+  ])
+
+  has_many(
+    :variants,
+    serializer: SnitchApiWeb.VariantView,
+    include: true
+  )
+
+  def variants(product, _conn) do
+    Map.get(product, :variants)
```

Comment:
Why was this required. Was it not coming from the schema. I might be missing something. Can you tell.

Reply:
That is just for safety. By default, it implements the function of every attribute. For an example, In our `Image` schema we have an attribute  called `url`. The `ja_serializer` defines the function `url/2` where there is alredy one defined from `conn` which gives conflicts. So, I renamed the attribute with `image_url` instead of `url` and overriding the `def image_url/2` function which extracts `url`.